internal/states/resource.go

```diff
@@ -29,7 +29,7 @@ type Resource struct {
 	// with a provider configuration when the resource configuration block is
 	// not available, such as if it has been removed from configuration
 	// altogether.
-	ProviderConfig addrs.AbsProviderConfig
+	ProviderConfig addrs.AbsProviderConfig // Ronny TODO: should we rename to ProviderResourceConfig? (this doesn't impact the state file)
```

Comment:
We can rename it to `ProviderResourceConfig` just to make sure we won't be confused with `ProviderInstanceConfig`

Reply:
I liked your idea https://github.com/opentofu/opentofu/pull/1963/commits/93aaf45228ccf0e9341cf03727b505249eeb2f1c